### PR TITLE
Update boot sequence docs

### PR DESCRIPTION
**Pull request by c1728p9**
Add content for the updated Mbed 5 boot sequence.

### CODE REVIEW

**Comment by c1728p9** [comment]
This has a code dependency on https://github.com/ARMmbed/mbed-os/pull/7794

File: docs/reference/runtime/Bootstrap.md [hunk 1]
```diff
@@ -7,13 +7,32 @@ Mbed OS provides two entry points for developers to hook into:
 - `main(void)` - Default entry point. All the standard application code goes here.
 - `mbed_main(void)` - Executed directly before `main`. The user can define this.
 
-When execution reaches the entry points, a user can expect a fully initialized system that is ready to execute application code. For this to happen, the following must have occurred prior to this point:
+When execution reaches the entry points, a user can expect a fully initialized system that is ready to execute application code. The Mbed OS boot sequence consists of four phases, target setup, toolchain setup, starting the RTOS and starting the Mbed application. These phases can be seen below:
```
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/reference/runtime/Bootstrap.md]: 👍 Yes, I like that. Neither one is wrong, but I think a colon is nicer here.

**Comment by AnotherButler** [comment]
ping @mprse

**Comment by AnotherButler** [comment]
FYI @MarceloSalazar @ashok-rao

**Pull request merged**